### PR TITLE
BaseJdbcClient and FTE test cleanup

### PR DESCRIPTION
**Pull request by hashhar**
## Description

- BaseJdbcClient cleanup
- Make FTE test concurrency customizable

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public BaseJdbcClient(
             Set<String> jdbcTypesMappedToVarchar,
             IdentifierMapping identifierMapping,
             RemoteQueryModifier remoteQueryModifier,
-            boolean supportsRetries)
+            @SupportsRetries boolean supportsRetries)
```
Comment (losipiuk) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java:118]: `BaseJdbcClient` is never directly instantiated (it is abstract) so this annotation should not be here.
The constructor should also probably be `protected`

Comment (hashhar) [reply plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java]: removed the entire commit

File: testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java [hunk 2]
```diff
@@ -77,21 +77,26 @@
...
-    private static final int MAX_PARALLEL_TEST_CONCURRENCY = 4;
 
     private final RetryPolicy retryPolicy;
-    private final Semaphore parallelTestsSemaphore = new Semaphore(MAX_PARALLEL_TEST_CONCURRENCY);
+    private final Semaphore parallelTestsSemaphore;
 
     protected BaseFailureRecoveryTest(RetryPolicy retryPolicy)
     {
         this.retryPolicy = requireNonNull(retryPolicy, "retryPolicy is null");
+        this.parallelTestsSemaphore = new Semaphore(getMaxParallelTestConcurrency());
```
Comment (kokosing) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java:87]: Calling a method that is going to be extended in a super constructor sounds like a java puzzle. Are you sure it is working the way you want? Maybe we can add an argument to the constructor?

Comment (losipiuk) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java:87]: Or you can move the logic to `@BeforeClass` method

**Review by losipiuk — APPROVED** [review]
LGTM % comment

File: testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java (hunk 2 above)
Comment (hashhar) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java]: extracted a parameter

**Comment by hashhar** [comment]
PTAL again @kokosing

**Review by wendigo — APPROVED** [review]
LGTM.

**Review by vlad-lyutenko — APPROVED** [review]

**Review by losipiuk — APPROVED** [review]

File: testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java [hunk 3]
```diff
@@ -81,14 +81,19 @@
...
     private static final Duration REQUEST_TIMEOUT = new Duration(5, SECONDS);
-    private static final int MAX_PARALLEL_TEST_CONCURRENCY = 4;
 
     private final RetryPolicy retryPolicy;
-    private final Semaphore parallelTestsSemaphore = new Semaphore(MAX_PARALLEL_TEST_CONCURRENCY);
+    private final Semaphore parallelTestsSemaphore;
 
     protected BaseFailureRecoveryTest(RetryPolicy retryPolicy)
+    {
+        this(retryPolicy, 4);
```
[the remark this answers is not in the document]
Comment (hashhar) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java]: done.

**Pull request merged**